### PR TITLE
Added Music and Sound Effects

### PR DESCRIPTION
**Pull request by saarthakchats**
#245 
http://rawgit.com/saarthakchats/labyrinth/master/index.html

Have added different music effects to four tiles in this PR.
Check "2018--new year" tile on the deploy link and the "you win" tile, and the "almost there" tile. Have added interesting sounds. Also the river tile.

### CODE REVIEW

**Comment by saarthakchats** [comment]
screenshot 27
Check it out!

File: js/tiles.js [hunk 1]
```diff
@@ -376,6 +377,7 @@ const door = {
         },
         /* Override the function */
         visit: function() {
+        	playAudio("outdoor.mp3");
```
Comment (abishekvashok) [inline js/tiles.js]: can you intend it with one more space

Comment (saarthakchats) [reply js/tiles.js]: Sure

**Comment by saarthakchats** [comment]
@abishekvashok

**Comment by saarthakchats** [comment]
Mentors, please don't merge yet. I and @samyakbvs are collaborating on this PR. So he will add more commits(sounds). Then merge, thanks.

File: js/tiles.js [hunk 2]
```diff
@@ -362,24 +362,26 @@ const door = {
...
+        	playAudio("suspense.mp3");
             alertInfo("Pretty close. Keep going!!", "");
             this.wallTop.show();
             this.wallRight.show();
             this.ground.show();
         },
     }),
     goal: Object.assign({}, OpenDoors, {
-        createImages: function() {
+            createImages: function() {
```
Comment (abishekvashok) [inline js/tiles.js]: un wanted change

File: js/tiles.js [hunk 3]
```diff
@@ -678,6 +681,7 @@ const door = {
             this.ground = this.createImage("tiles/rooms/floor/floorNewYear.svg");
         },
         visit: function(player) {
+        	playAudio("celebrate.mp3");
```
Comment (abishekvashok) [inline js/tiles.js]: need one more space

File: js/tiles.js [hunk 4]
```diff
@@ -541,6 +543,7 @@ const door = {
             this.ground = this.createImage("tiles/rooms/floor/river.svg");
         },
         visit: function() {
+        	playAudio("watersound.mp3");
```
Comment (abishekvashok) [inline js/tiles.js]: one more space

File: js/tiles.js [hunk 5]
```diff
@@ -362,24 +362,26 @@ const door = {
...
             this.ground = this.createImage("tiles/rooms/floor/goal.svg");
         },
         /* Override the function */
         visit: function() {
-            alertSuccess("You win!", "Yay! You have won the game");
-            this.wallTop.show();
-            this.wallRight.show();
-            this.ground.show();
+        	 playAudio("outdoor.mp3");
+             alertSuccess("You win!", "Yay! You have won the game");
```
Comment (abishekvashok) [inline js/tiles.js]: pls reduce a space

File: js/tiles.js [hunk 6]
```diff
@@ -362,24 +362,26 @@ const door = {
...
+             this.wallTop.show();
```
Comment (abishekvashok) [inline js/tiles.js]: here as well

File: js/tiles.js [hunk 7]
```diff
@@ -362,24 +362,26 @@ const door = {
...
+             this.wallRight.show();
```
Comment (abishekvashok) [inline js/tiles.js]: here

File: js/tiles.js [hunk 8]
```diff
@@ -362,24 +362,26 @@ const door = {
...
+             this.ground.show();
```
Comment (abishekvashok) [inline js/tiles.js]: and here

**Comment by saarthakchats** [comment]
Still don't merge, need to wait for @samyakbvs

**Comment by samyakbvs** [comment]
I will be doing the changes in sometime

**Comment by niccokunzmann** [comment]
I need to know that these sounds are ok to be AGPL licensed. Please provide a proof.

**Comment by samyakbvs** [comment]
@saarthakchats Pls merge my PR ;)

**Comment by niccokunzmann** [comment]
Please show that it is legal to license the music under AGPL. If not, please provide the license of the music and we can look further.

**Comment by niccokunzmann** [comment]
Did you create the music?

**Comment by samyakbvs** [comment]
Nope @niccokunzmann

**Comment by samyakbvs** [comment]
https://archive.org/download/brennanworldwide5sec This is the site from which I took the tunes

**Comment by 0x48piraj** [comment]
I'm good at digital beat boxing.
I can create some beats, but they'll be somewhat not good as this (I guess)

**Comment by niccokunzmann** [comment]
@0x48piraj This is great. I would like to solve the issue #245 to have this work well. It is about community in my eyes.

**Comment by niccokunzmann** [comment]
@saarthakchats Thanks. I see https://ia600602.us.archive.org/21/items/brennanworldwide5sec/brennanworldwide5sec_meta.xml it is licensed under creative commons no derivatives. Please see #291 and document the license. Then, I can merge it without doubt!

**Comment by niccokunzmann** [comment]
I would also like to improve #291 so if you leave a review, it is much appreciated.

**Comment by saarthakchats** [comment]
Will add license and reference in another PR right after this. Have seen #291 
Will add my license and website I got music from to that "license" folder immediately after this is merged. Pls, merge this so I can add that. I will do it (btw I took it from https://soundbible.com licensed under creative commons)  Please merge this
@niccokunzmann

**Comment by saarthakchats** [comment]
@niccokunzmann 
Pls merge

**Comment by saarthakchats** [comment]
@niccokunzmann 
I have added the music License. Please merge now

**Comment by arjunpandey2003** [comment]
@niccokunzmann under the same license I have made my PR. https://github.com/fossasia/labyrinth/pull/296

**Comment by saarthakchats** [comment]
@niccokunzmann Pls merge
License is added

File: js/tiles.js [hunk 9]
```diff
@@ -694,10 +709,11 @@ const door = {
             this.ground = this.createImage("tiles/rooms/floor/floorNewYear.svg");
         },
         visit: function(player) {
-            this.wallTop.show();
-            this.wallRight.show();
-            this.ground.show();
-            alertNormal("Happy 2018!', 'It's new year already! Don't waste your time and explore!", "");
+        	  playAudio("celebrate.mp3");
```
Comment (anshumanv) [inline js/tiles.js]: Indent this and please squash commits, rest LGTM

**Comment by saarthakchats** [comment]
@niccokunzmann 
Pls merge

**Comment by niccokunzmann** [comment]
@saarthakchats Please do the following: add

- [ ] audio/celebrate.mp3.license
- [ ]  audio/confusion.mp3.license
- [ ] audio/discover.mp3.license
- [ ] audio/gears.mp3.license
- [ ] audio/spy.mp3.license
- [ ] audio/suspense.mp3.license
- [ ] audio/watersound.mp3.license

See https://github.com/fossasia/labyrinth#how-to-add-music

**Comment by saarthakchats** [comment]
@niccokunzmann 
I will add those by creating an issue immediately after this is merged, pls merge this. Thanks

**Comment by niccokunzmann** [comment]
Thanks for this. Also thanks for resolving the conflict :)

**Comment by niccokunzmann** [comment]
Added issue https://github.com/fossasia/labyrinth/issues/346 for the licensing.

**Pull request merged**